pkg/virtctl/create/preference/preference_test.go

```diff
@@ -140,20 +146,44 @@ func setFlag(flag, parameter string) string {
 	return fmt.Sprintf("--%s=%s", flag, parameter)
 }
 
-func getPreferenceSpec(bytes []byte, namespaced bool) (*instancetypev1beta1.VirtualMachinePreferenceSpec, error) {
+func testAdmissionAndGetSpec(bytes []byte, namespaced bool) (*instancetypev1beta1.VirtualMachinePreferenceSpec, error) {
 	decodedObj, err := runtime.Decode(generatedscheme.Codecs.UniversalDeserializer(), bytes)
 	Expect(err).ToNot(HaveOccurred())
 
 	switch obj := decodedObj.(type) {
 	case *instancetypev1beta1.VirtualMachinePreference:
 		Expect(namespaced).To(BeTrue())
 		Expect(strings.ToLower(obj.Kind)).To(Equal(instancetype.SingularPreferenceResourceName))
+		Expect(testAdmission(obj, instancetype.PluralPreferenceResourceName, &admitters.PreferenceAdmitter{})).To(BeTrue())
 		return &obj.Spec, nil
 	case *instancetypev1beta1.VirtualMachineClusterPreference:
 		Expect(namespaced).To(BeFalse())
 		Expect(strings.ToLower(obj.Kind)).To(Equal(instancetype.ClusterSingularPreferenceResourceName))
+		Expect(testAdmission(obj, instancetype.ClusterPluralPreferenceResourceName, &admitters.ClusterPreferenceAdmitter{})).To(BeTrue())
 		return &obj.Spec, nil
 	}
 
 	return nil, fmt.Errorf("object must be VirtualMachinePreference or VirtualMachineClusterPreference")
 }
+
+func testAdmission(obj runtime.Object, resource string, admitter validating_webhooks.Admitter) bool {
```

Comment:
Admission controllers is a server-side concept. I don't expect a command line tool to care how the API is validated, be it an admission controller or inside the reconcile loop.

I think that it would be nicer to have a function such as `IsClusterPreferenceValid` and call it both from here and from the admission controller.

Reply:
I reworked this to be less dependent on the Admitters interface. It is using `ValidateInstanceTypeSpec` and `ValidatePreferenceSpec` now to validate the specs of created objects. IMO this comes as close as possible to your suggestion without reworking too many parts of the code base. Extracting the validation logic further could be a follow up.

Reply:
> Extracting the validation logic further could be a follow up.

Thanks. Having `"kubevirt.io/kubevirt/pkg/virt-api/webhooks/validating-webhook/admitters"` imported by the client is still a "code smell".